corehq/motech/repeaters/repeater_generators.py

```diff
@@ -485,10 +517,23 @@ def get_case_index(self, target_case):
         if index_case.case_type != self.index_create_case_type:
             raise DataRegistryCaseUpdateError("Index case type does not match")
 
-        relationship = self.index_create_relationship or "child"
-        key = "parent" if relationship == "child" else "host"
         return {
-            key: (self.index_create_case_type, self.index_create_case_id, relationship)
+            self.index_create_identifier: (
+                self.index_create_case_type, self.index_create_case_id, self.index_create_relationship
+            )
+        }
+
+    def get_remove_case_index(self, target_case):
+        identifier = "parent" if self.index_remove_relationship == "child" else "host"
+        indices = [index for index in target_case.live_indices if index.identifier == identifier]
+        if not indices:
+            return {}
```

Comment:
The user's index removal attempt will be rejected if they provide the wrong `referenced_id` - should it also fail if they attempt to remove an index that doesn't exist?  (Honest question, not a suggestion)

Reply:
I did think about that but decided not to fail in that instance. I guess it depends how strict we want to be. I can follow up with the delivery team to check.

Reply:
Followed up on this and decided to leave this as is.